skbio/metadata/_interval.py

```diff
@@ -559,6 +559,58 @@ def _reverse(self):
         # DONT' forget this!!!
         self._is_stale_tree = True
 
+    @classonlymethod
+    @experimental(as_of="0.5.0-dev")
+    def concat(cls, intvl_metadata):
+        '''Concatenate an iterable of ``IntervalMetadata`` objects.
+
+        Parameters
+        ----------
+        intvl_metadata : Iterable (IntervalMetadata)
+
+        Returns
+        -------
+        IntervalMetadata
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> im1 = IntervalMetadata(3)
+        >>> _ = im1.add([(0, 2)], [(True, False)], {'gene': 'sagA'})
+        >>> im2 = IntervalMetadata(4)
+        >>> _ = im2.add([(1, 4)], [(True, True)], {'gene': 'sagB'})
+        >>> IntervalMetadata.concat([im1, im2])  # doctest: +ELLIPSIS
+        2 interval features
+        -------------------
+        Interval(interval_metadata=<...>, bounds=[(0, 2)], \
+fuzzy=[(True, False)], metadata={'gene': 'sagA'})
+        Interval(interval_metadata=<...>, bounds=[(4, 7)], \
+fuzzy=[(True, True)], metadata={'gene': 'sagB'})
+
+        '''
+        intvl_metadata = list(intvl_metadata)
+
+        if len(intvl_metadata) == 0:
+            return cls(0)
+
+        upper_bound = 0
+        for im in intvl_metadata:
+            upper_bound += im.upper_bound
+        new = cls(upper_bound)
+
+        length = 0
+        for i, im in enumerate(intvl_metadata):
+            for intvl in im._intervals:
+                bounds = intvl.bounds
+                fuzzy = intvl.fuzzy
+                if i != 0:
+                    bounds = [(start + length, end + length)
```

Comment:
What's the method for merging or concatenating (unsure of the best verb here) for overlapping intervals?